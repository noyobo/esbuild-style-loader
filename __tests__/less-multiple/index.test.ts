import path from 'path';
import { runTest } from '../runTest';
import fse from 'fs-extra';

describe('less-multiple', function () {
  it(path.basename(__filename), async function () {
    const output = path.resolve(__dirname, 'output');
    fse.removeSync(output);
    const result = await runTest([path.resolve(__dirname, './index.js')], output);
    const allFile = result.outputFiles.map((item) => item.path);
    allFile.forEach((file) => {
      fse.ensureFileSync(file);
      const content = result.outputFiles.find((item) => item.path === file)?.text;
      fse.writeFileSync(file, content);
      expect(content).toMatchSnapshot(path.relative(process.cwd(), file));
    });
  });

  it(path.basename(__filename), async function () {
    const output = path.resolve(__dirname, 'output-2');
    fse.removeSync(output);
    const result = await runTest([path.resolve(__dirname, './index.js')], output, {
      bundle: false,
    });
    const allFile = result.outputFiles.map((item) => item.path);
    allFile.forEach((file) => {
      fse.ensureFileSync(file);
      const content = result.outputFiles.find((item) => item.path === file)?.text;
      fse.writeFileSync(file, content);
      expect(content).toMatchSnapshot(path.relative(process.cwd(), file));
    });
  });
});
