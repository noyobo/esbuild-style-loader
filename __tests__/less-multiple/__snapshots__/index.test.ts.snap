// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`less-multiple index.test.ts: __tests__/less-multiple/output/index.css 1`] = `
"/* css-loader:./a.less */
.a__a {
  color: red;
}

/* css-loader:./b.less */
.b__b {
  color: green;
}
/*# sourceMappingURL=index.css.map */
"
`;

exports[`less-multiple index.test.ts: __tests__/less-multiple/output/index.css.map 1`] = `
"{
  "version": 3,
  "sources": ["css-loader:./a.less", "css-loader:./b.less"],
  "sourcesContent": [".a__a {\\n  color: red;\\n}\\n\\n/* #sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VSb290IjpudWxsLCJtYXBwaW5ncyI6IkFBQUEiLCJzb3VyY2VzIjpbIlVzZXJzL25veW9iby9ob21lL2dpdGh1Yi9lc2J1aWxkLXN0eWxlLWxvYWRlci9fX3Rlc3RzX18vbGVzcy1tdWx0aXBsZS9hLmxlc3MiXSwic291cmNlc0NvbnRlbnQiOlsiLmEge1xuICBjb2xvcjogcmVkO1xufVxuIl0sIm5hbWVzIjpbXX0=*/", ".b__b {\\n  color: green;\\n}\\n\\n/* #sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VSb290IjpudWxsLCJtYXBwaW5ncyI6IkFBQUEiLCJzb3VyY2VzIjpbIlVzZXJzL25veW9iby9ob21lL2dpdGh1Yi9lc2J1aWxkLXN0eWxlLWxvYWRlci9fX3Rlc3RzX18vbGVzcy1tdWx0aXBsZS9iLmxlc3MiXSwic291cmNlc0NvbnRlbnQiOlsiLmIge1xuICBjb2xvcjogZ3JlZW47XG59XG4iXSwibmFtZXMiOltdfQ==*/"],
  "mappings": ";AAAA;AACE;AAAA;;;ACDF;AACE;AAAA;",
  "names": []
}
"
`;

exports[`less-multiple index.test.ts: __tests__/less-multiple/output/index.js 1`] = `
"(() => {
  // style-loader:./a.less
  var a_default = { "a": "a__a" };

  // style-loader:./b.less
  var b_default = { "b": "b__b" };

  // __tests__/less-multiple/index.js
  console.log(a_default, b_default);
})();
//# sourceMappingURL=index.js.map
"
`;

exports[`less-multiple index.test.ts: __tests__/less-multiple/output/index.js.map 1`] = `
"{
  "version": 3,
  "sources": ["style-loader:./a.less", "style-loader:./b.less", "../index.js"],
  "sourcesContent": ["import './a.less?modules';\\nexport default {\\"a\\":\\"a__a\\"};", "import './b.less?modules';\\nexport default {\\"b\\":\\"b__b\\"};", "import a from './a.less?modules';\\nimport b from './b.less?modules';\\n\\nconsole.log(a, b);\\n"],
  "mappings": ";;AACA,MAAO,YAAQ,EAAC,KAAI,OAAM;;;ACA1B,MAAO,YAAQ,EAAC,KAAI,OAAM;;;ACE1B,UAAQ,IAAI,WAAG,SAAC;",
  "names": []
}
"
`;
